modules/schema/src/main/java/org/apache/ignite/internal/schema/marshaller/reflection/ObjectStatistics.java

```diff
@@ -23,35 +23,81 @@
 import org.apache.ignite.internal.marshaller.Marshaller;
 import org.apache.ignite.internal.schema.Column;
 import org.apache.ignite.internal.schema.SchemaDescriptor;
+import org.apache.ignite.internal.schema.marshaller.MarshallerUtil;
 import org.apache.ignite.internal.schema.row.RowAssembler;
 import org.apache.ignite.internal.type.NativeType;
 import org.jetbrains.annotations.Nullable;
+import org.jetbrains.annotations.TestOnly;
 
 /**
  * Object statistic.
  */
 class ObjectStatistics {
-    /** Cached zero statistics. */
-    private static final ObjectStatistics ZERO_STATISTICS = new ObjectStatistics(-1);
-
     /** Estimated total size of the object. */
     private final int estimatedValueSize;
 
+    private final Object[] values;
+
     /** Constructor. */
-    private ObjectStatistics(int estimatedValueSize) {
+    private ObjectStatistics(Object[] values, int estimatedValueSize) {
+        this.values = values;
         this.estimatedValueSize = estimatedValueSize;
     }
 
-    private int getEstimatedValueSize() {
+    int estimatedValueSize() {
         return estimatedValueSize;
     }
 
+    @TestOnly
+    <T> T value(int index) {
+        return (T) values[index];
+    }
+
+    static MarshallerRowBuilder createRowBuilder(SchemaDescriptor schema, Marshaller keyMarsh, Object key) {
```

Comment:
Expression like `ObjectStatistics.createRowBuilder(...)` looks weird. 

Let's move factory methods to MarshallerRowBuilder class or MarshallerUtil.

Reply:
Moved to `MarshallerRowBuilder`, thanks.